### DISCUSSION TITLE
Why docker?

### DISCUSSION THREAD

**Discussion by buhtz** [initialcommit-com/git-sim#87] [Q&A] [answered]
I'm just asking in favour to learn something. ;)

I'm not a docker user. I know it makes sense in CI processes etc.

But why should someone use such a "simple" application like git-sim in a docker? What is the advantage?

**Comment by abhijitnathwani** [comment, marked as answer]
Hi @buhtz 
The idea was to keep dependencies bundled for people who want to keep things separately. Without having to install all the dependencies, the docker image keeps things ready. 

Also, the community asked for it. See: https://github.com/initialcommit-com/git-sim/issues/46

**Reply by buhtz** [reply]
The initial post in #46 just shows me someone who is not able to handle his Python environment. IMHO using docker here is like driving your Ferrari to the backery on the next corner to buy buns. :D What you really need is distro packages in Debian and Arch. But that isn't your job as main straim. But you could have directed your users in that direction. And OS X users? They can prey to Mr. Jobs if the have problems with Python on OS X.

**Reply by initialcommit-io** [reply]
@buhtz The only reason we implemented the dockerfile is because we got many user requests for it.  Is it overkill for a small tool like Git-Sim? Probably.  But the main annoyance with current Git-Sim installation process (which you hinted at) is that some Manim dependencies must be installed manually. This makes it a multistep process that some users might just want to forget about and just use docker.

I'd love to get some binaries created that would enable a standalone program with a 1-step installation process.  I don't have much experience with that so it's something in the works.

**Reply by buhtz** [reply]
I just installed via `python3 -m pip install git-sim`. Worked well on Debian 11. No manual Manim installation needed. Why not integrate Manim in the python packaing process (via pyproject.toml (and/or setup.py; or some other Python magic)).
